src/check_executor.rs

```diff
@@ -170,11 +205,27 @@ async fn executor_loop(
                         job_checker.check_url(config, tick, &job_region).await
                     };
 
+                    let will_retry = check_result.status == CheckStatus::Failure
+                        && scheduled_check.retry_count < conf.failure_retries;
+
+                    record_result_metrics(
+                        &check_result,
+                        scheduled_check.retry_count > 0,
+                        will_retry,
+                    );
+
+                    // re-queue for execution again
+                    if will_retry {
+                        tracing::debug!(result = ?check_result, "executor.check_will_retry");
+                        job_check_sender.queue_check_for_retry(scheduled_check);
+                        job_num_running.fetch_sub(1, Ordering::SeqCst);
```

Comment:
Let's move this out of the if statement, and just put it before `will_retry` so that we're not duplicating?

Reply:
👍

Reply:
Actally I'm going to leave this here, just since we also do the sending with the executor. so if we move it up we'll be skipping the sending bit and the metric won't be quite accurate